src/main/java/net/dv8tion/jda/client/managers/ApplicationManagerUpdatable.java

```diff
@@ -0,0 +1,366 @@
+        {
+            @Override
+            public void checkValue(final List<String> value)
+            {
+                Args.notNull(value, "redirect uris");
+                for (final String url : value)
+                {
+
+                    Args.notNull(url, "redirect uri");
+                    if (!ApplicationManagerUpdatable.URL_PATTERN.matcher(url).matches())
+                        throw new IllegalArgumentException("URL must be a valid http or https url.");
+                }
+            }
+            @Override
+            public ApplicationManagerUpdatable setValue(List<String> value)
+            {
+                checkValue(value);
+
+                this.value = Collections.unmodifiableList(new ArrayList<>(value));
+                this.set = true;
+
+                return manager;
+            }
+        };
+    }
+
+    /**
+     * Creates a new {@link net.dv8tion.jda.core.requests.RestAction RestAction} instance
+     * that will apply <b>all</b> changes that have been made to this manager instance (one per runtime per JDA instance).
+     *
+     * <p>Before applying new changes it is recommended to call {@link #reset()} to reset previous changes.
+     * <br>This is automatically called if this method returns successfully.
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} - Type: Void
+     *         <br>Updates all modified fields or does nothing if none of the {@link net.dv8tion.jda.core.managers.fields.Field Fields}
+     *         have been modified. ({@link net.dv8tion.jda.core.requests.RestAction.EmptyRestAction EmptyRestAction})
+     */
+    public RestAction<Application> update()
+    {
+        if (!this.needsUpdate())
+            return new RestAction.EmptyRestAction<>(this.application);
+
+        final JSONObject body = new JSONObject();
+
+        // All fields are required or they are resetted to default
+
+        body.put("description",
+                this.description.shouldUpdate() ? this.description.getValue() : this.description.getOriginalValue());
+
+        body.put("bot_require_code_grant", this.doesBotRequireCodeGrant.shouldUpdate()
+                ? this.doesBotRequireCodeGrant.getValue() : this.doesBotRequireCodeGrant.getOriginalValue());
+
+        body.put("icon",
+                this.icon.shouldUpdate()
+                        ? this.icon.getValue() == null ? JSONObject.NULL : this.icon.getValue().getEncoding()
+                        : this.application.getIconUrl());
+
+        body.put("bot_public",
+                this.isBotPublic.shouldUpdate() ? this.isBotPublic.getValue() : this.isBotPublic.getOriginalValue());
+
+        body.put("name", this.name.shouldUpdate() ? this.name.getValue() : this.name.getOriginalValue());
+
+        body.put("redirect_uris", this.redirectUris.shouldUpdate() ? this.redirectUris.getValue() :  this.redirectUris.getOriginalValue());
```

Comment:
You shouldn't include fields that are not updated here, especially not for icon since this will most definitely not work here.
```java
if (this.description.shouldUpdate())
    body.put("description", this.description.getValue());
```